crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -8841,6 +8845,19 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
         if let Type::SpecialForm(SpecialFormType::Tuple) = value_ty {
             return tuple_generic_alias(self.db(), self.infer_tuple_type_expression(slice));
         }
+        if let Type::KnownInstance(KnownInstanceType::TypeAliasType(TypeAliasType::PEP695(
+            type_alias,
+        ))) = value_ty
```

Comment:
I'd suggest adding a `TODO` comment that the limitation to PEP 695 type aliases is temporary, and only because that's the only kind of type alias that we've attached a generic context to yet.